### DISCUSSION TITLE
Atlascope Phase 1 Review

### DISCUSSION THREAD

**Discussion by annehaley** [atlascope/atlascope#210] [Ideas]
This is an incomplete list of feature analyses that can act as a starter for Phase 2 design discussions.

**Design that works**

- Dataset embeddings: The semantic zoom works well. We may run into data cases where our model does not fit, but it is adaptable enough for now.

- Importer structure: Our current design with builtin importer modules works far better than any plugin structure could. We're able to consistently import data from Vandy, Google Drive, or a local path.

- Job infrastructure: The backend portion of this is satisfactory, but the frontend (UX) could be improved, as well as the communication between them about the state of a  spawned job.

**Design to reconsider**

- Serving tiles of large files in a _remote_ location: We'll need to reapproach this problem once the solution of storing files locally is no longer an option.

- The Pin model: What are the use cases for these and how do they differ?
We overloaded this model by using them to give a location to job results (unnecessary). Pins, in theory, should have a location and some information. In their original concept, Pins were meant to be a method of organizing knowledge obtained from the data so that conclusions can be drawn. The interface for interacting with / organizing / editing / creating Pins should be more robust in Phase 2.

- Subclassing the Dataset model? We may be overloading this model as well. We have a field called `dataset_type` that can be any of the following: `['tile_source', 'tile_overlay', 'analytics', 'subimage', 'structure_detection', 'non_tiled_image', '3d_volume' ]`. Some of these have completely different behaviors, so it may be a good idea to separate them.

- User experience: Rather than pushing nearly all functionality into a tabbed sidebar, we can consult with our Phase 2 customer and/or our UX team to find a layout that obscures less from the user.

- Users on Investigations: This was something we explored in the beginning and decided was unnecessary for Phase 1. In a more robust system, users would be required for access control on investigations and datasets.

- At one point we discussed a model for "Scale". Could this be necessary / helpful in the future?

**Comment by waxlamp** [comment]
Good stuff, @annehaley, thank you!
